pkg/repository/prisma/event.go

```diff
@@ -194,6 +194,24 @@ func (r *eventAPIRepository) ListEventKeys(tenantId string) ([]string, error) {
 	return keys, nil
 }
 
+func sizeOfEvent(item *repository.CreateEventOpts) int {
+	return len(item.Data) + len(item.AdditionalMetadata)
+}
+
+func (e *eventEngineRepository) StartBufferLoop() (func() error, error) {
```

Comment:
If the buffer loop is absolutely necessary for the event repository to function as expected, then we shouldn't expose it as a function to any callers. We should start it on `NewEventEngineRepository` and return a cleanup method out of the EngineRepository, otherwise we're doing a lot of `StartBufferLoop` between packages when it doesn't seem necessary.

Reply:
makes sense, passing the cleanup and errors back up the chain is the annoying bit, I've modified the NewEventRepo calling to return the cleanup as part of it's cleanup and the error as an additional return value

Reply:
added a Cleanup method to EventEngineRepository to take care of cleanup